be/src/storage/rowset/dictcode_column_iterator.cpp

```diff
@@ -94,9 +127,14 @@ Status GlobalDictCodeColumnIterator::build_code_convert_map(ScalarColumnIterator
     return Status::OK();
 }
 
-ColumnPtr GlobalDictCodeColumnIterator::_new_local_dict_col(bool nullable) {
+ColumnPtr GlobalDictCodeColumnIterator::_new_local_dict_col(Column* src) {
     ColumnPtr res = std::make_unique<Int32Column>();
-    if (nullable) {
+    auto code_data = ColumnHelper::get_data_column(src);
+    if (code_data->is_array()) {
+        res = ArrayColumn::create(NullableColumn::create(res, NullColumn::create()), UInt32Column::create());
```

Comment:
Is this an implicit constraint?

Reply:
I think not, dict-iterator read interface must keep same as normal-iterator interface